File: gypsum/colors.py
# Prints out all 16 colors (bright & dark)
# against each other as both the foreground
# and background colors

text = ' gYiw ' # the test text

colors = {
    'color0': { 'fg': '30', 'bg': '40' },
    'color1': { 'fg': '31', 'bg': '41' },
    'color2': { 'fg': '32', 'bg': '42' },
    'color3': { 'fg': '33', 'bg': '43' },
    'color4': { 'fg': '34', 'bg': '44' },
    'color5': { 'fg': '35', 'bg': '45' },
    'color6': { 'fg': '36', 'bg': '46' },
    'color7': { 'fg': '37', 'bg': '47' },
    'color8': { 'fg': '90', 'bg': '100'},
    'color9': { 'fg': '91', 'bg': '101'},
    'color10': { 'fg': '92', 'bg': '102'},
    'color11': { 'fg': '93', 'bg': '103'},
    'color12': { 'fg': '94', 'bg': '104'},
    'color13': { 'fg': '95', 'bg': '105'},
    'color14': { 'fg': '96', 'bg': '106'},
    'color15': { 'fg': '97', 'bg': '107'},
}

names = {
    'black': {
        'dark': 'color0',
        'bright': 'color8'
    },
    'red': {
        'dark': 'color1',
        'bright': 'color9'
    },
    'green': {
        'dark': 'color2',
        'bright': 'color10'
    },
    'yellow': {
        'dark': 'color3',
        'bright': 'color11'
    },
    'blue': {
        'dark': 'color4',
        'bright': 'color12'
    },
    'magenta': {
        'dark': 'color5',
        'bright': 'color13'
    },
    'cyan': {
        'dark': 'color6',
        'bright': 'color14'
    },
    'white': {
        'dark': 'color7',
        'bright': 'color15'
    }
}


lines = []
columns = []
columns_done = False
for fg_name in names:
    for fg_shade in ['dark', 'bright']:
        fg_id = names[fg_name][fg_shade]
        fg = colors[fg_id]['fg']
        line = []
        line.append(f'{fg_shade} {fg_name}')
        line.append(f' {fg_id} '.ljust(9))
        for bg_name in names:
            for bg_shade in ['dark', 'bright']:
                bg_id = names[bg_name][bg_shade]
                bg = colors[bg_id]['bg']
                line.append(f'\033[{fg}m\033[{bg}m{text}\033[0m')
                if not columns_done:
                    bg_id_num = bg_id[-2:]
                    if bg_id_num[0] == 'r':
                        columns.append(f' {bg_shade[0]}{bg_name[0]}{bg_id_num[1]}  ')
                    else:
                        columns.append(f' {bg_shade[0]}{bg_name[0]}{bg_id_num} ')
        columns_done = True
        lines.append(line)

max_left_col_len = max([len(i[0]) for i in lines])
for line in lines:
    line[0] = line[0].center(max_left_col_len)

header = ''
header += 'foreground \\ background '.center(len(lines[0][0] + lines[0][1]))
header += '|'
header += '|'.join(columns)

print()
print(header)
print(''.join(['-']*len(header)))
print('\n'.join([' '.join(i) for i in lines]))
print()

File: gypsum/.config/nvim/lua/haondt/colors.lua
local M = {}
M.colors = {
    -- special
    none = "NONE",
    bg = "#0b0708",
    fg = "#f3c6b1",

    -- black
    color0 = "#8e6c51",
    color8 = "#b89f8b",

    -- red
    color1 = "#d44e45",
    color9 = "#cc6666",

    -- green
    color2 = "#4f5040",
    color10 = "#7daf69",

    -- yellow
    color3 = "#fcb685",
    color11 = "#fcd1b3",

    -- blue
    color4 = "#c46692",
    color12 = "#dfa0bd",

    -- magenta
    color5 = "#8d479e",
    color13 = "#ad73be",

    -- cyan
    color6 = "#d98887",
    color14 = "#f3afaf",

    -- white
    color7 = "#3e2d34",
    color15 = "#eacfda",
}

M.palette = {
    none = M.colors.none,
    bg = M.colors.none,
    fg = M.colors.fg,
    color0 = M.colors.color0,
    color1 = M.colors.color1,
    color2 = M.colors.color2,
    color3 = M.colors.color3,
    color4 = M.colors.color4,
    color5 = M.colors.color5,
    color6 = M.colors.color6,
    color7 = M.colors.color7,
    color8 = M.colors.color8,
    color9 = M.colors.color9,
    color10 = M.colors.color10,
    color11 = M.colors.color11,
    color12 = M.colors.color12,
    color13 = M.colors.color13,
    color14 = M.colors.color14,
    color15 = M.colors.color15,

    selected = { bg = M.colors.color7 },
    popup_border = { fg = M.colors.color13 },
    search = { bg = M.colors.color9, fg = M.colors.color7 },

    error = { fg = M.colors.color15, bg = M.colors.color1 },
    add = { fg = "#7daf69" },
    remove = {fg = M.colors.color9 },
}

return M
